[ci/docs] add some documentation about the stats uploading process

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #79504

This process is pretty confusing, so wrote it down.

[skip ci]

suo committed:
[ci/docs] add some documentation about the stats uploading process

This process is pretty confusing, so wrote it down.

[skip ci]

[ghstack-poisoned]

janeyx99 commented on tools/stats/README.md:
```diff
@@ -0,0 +1,31 @@
+   https://github.com/pytorch/pytorch/blob/a9f6a35a33308f3be2413cc5c866baec5cfe3ba1/.github/workflows/_linux-build.yml#L144-L151
+2. When a workflow completes, a `workflow_run` event [triggers
+   `upload-test-stats.yml`](https://github.com/pytorch/pytorch/blob/d9fca126fca7d7780ae44170d30bda901f4fe35e/.github/workflows/upload-test-stats.yml#L4).
+3. `upload-test-stats` downloads the raw stats form the intermediate data store
```
```suggestion
3. `upload-test-stats` downloads the raw stats from the intermediate data store
```

janeyx99 reviewed (approved):
yay

suo committed:
Update on "[ci/docs] add some documentation about the stats uploading process"


This process is pretty confusing, so wrote it down.

[skip ci]

[ghstack-poisoned]

suo commented:
@pytorchbot merge -f
